### DISCUSSION TITLE
Rowy Config files?

### DISCUSSION THREAD

**Discussion by runnedrun** [AntlerVC/firetable#703] [Support / Q&A] [answered]
Hi, love the idea of Rowy and hope my dev shop can start using it on our projects! The main feature missing for us right now is (as far as I can tell) the ability to programmatically configure Rowy. Does this functionality already exist? For instance, right now we generate basic admin tables for clients from typescript types (that themselves are turned into JSON schemas). This ensures that our admin panels are always up to date with our code AND allows us to easily configure and modify our admin panels with code that is in git (as opposed to somebody randomly configuring some UI and not knowing how to undo it). 

Does Rowy have any way for us to use code to configure the tables, fields, etc in a sheet? (without needing to dig into the source)

Thanks!

**Comment by harinij** [comment, marked as answer]
Hi @runnedrun 👋 We currently have an option that lets you import/export a table setting as a json (short video on how to below). Also, only admins of the project can create tables, add columns etc. That way other roles invited to the project are prevented from making UI changes.

https://user-images.githubusercontent.com/307298/178686874-b58b1d7a-1cf6-4492-979e-517a1db7dea1.mov

All the config of a project is stored under `_rowy_/settings` even though it is not recommended to do direct manipulation of that config, that's another option if needed.

PS: we are bringing multiple projects/workspaces support soon so watch this space :)

**Reply by runnedrun** [reply]
OK great, thanks for the quick response @harinij . Follow up question: Is there a way to do this programmatically? Seems like we can generate the importable json from our TS types, but is there an endpoint we can hit to automatically setup new or update existing Rowy tables? 

Related, is it possible to import json to update an existing table?

Thanks!

**Reply by notsidney** [reply]
Hey, you can see your Rowy config in your Firestore database under `_rowy_/settings` where there is a `tables` array. You can find the type for that here: https://github.com/rowyio/rowy/blob/e4e19f3a9214bcca99beeb4f52d0f358df66aab3/src/types/table.d.ts#L63-L80


Then, to add columns, you can create or edit documents in `_rowy_/settings/schema/<table_id>`. The type for that document is:
https://github.com/rowyio/rowy/blob/e4e19f3a9214bcca99beeb4f52d0f358df66aab3/src/types/table.d.ts#L82-L98

**Reply by runnedrun** [reply]
Amazing, this is great, thanks!
